app/modules/parsing/graph_construction/parsing_helper.py

```
@@ -1639,26 +1689,39 @@ async def _clone_to_repo_manager(
+                logger.info(f"ParsingHelper: Successfully cloned {repo_name} as bare repo")
+
+                # Configure the bare repo to fetch all refs
+                bare_repo = Repo(str(bare_repo_path))
+                if bare_repo.remotes:
+                    origin = bare_repo.remotes.origin
+                    origin.fetch()
+                    logger.info(f"ParsingHelper: Fetched all refs for {repo_name}")
+
             except Exception as e:
-                logger.exception(f"Failed to clone {repo_name}: {e}")
+                logger.exception(f"Failed to clone {repo_name} as bare repo: {e}")
```

Ruff (0.15.1)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)